sdk/src/main/java/ly/count/android/sdk/ModuleCrash.java

```diff
@@ -142,6 +142,43 @@ public void sendCrashReportToQueue(String error, boolean nonfatal, boolean isNat
         requestQueueProvider.sendCrashReport(crashData, nonfatal);
     }
 
+    public void sendCrashReportToQueueWFilterCallback(String error, boolean nonfatal, boolean isNativeCrash, @Nullable final Map<String, Object> customSegmentation) {
+        L.d("[ModuleCrash] sendCrashReportToQueueWFilterCallback");
+
+        Map<String, Object> combinedSegmentationValues = new HashMap<>();
+
+        if (customCrashSegments != null) {
+            combinedSegmentationValues.putAll(customCrashSegments);
+        }
+
+        if (customSegmentation != null) {
+            Utils.removeUnsupportedDataTypes(customSegmentation);
+            combinedSegmentationValues.putAll(customSegmentation);
+        }
+
+        //truncate crash segmentation
+        Utils.truncateSegmentationValues(combinedSegmentationValues, _cly.config_.maxSegmentationValues, "[ModuleCrash] sendCrashReportToQueueWFilterCallback", L);
+
+        //limit the size of the crash report to 20k characters
+        if (!isNativeCrash) {
+            error = error.substring(0, Math.min(20_000, error.length()));
+        }
+
+        CrashData crashData = new CrashData(error, combinedSegmentationValues, DeviceInfo.getLogs());
+
+        if(globalCrashFilterCallback != null) {
+            if (globalCrashFilterCallback.filterCrash(crashData)) {
+                L.d("[ModuleCrash] Crash filter found a match, exception will be ignored, [" + error.substring(0, Math.min(error.length(), 60)) + "]");
+                return;
+            }
+        }
```

Comment:
if the crash data was modiffied, then in crash metrics we need to set a custom flag